### PR TITLE
Update changelog for 1.4.0

### PR DESCRIPTION
**Pull request by yophilav**
Update changelog for 1.4.0

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [x] concise summary of tests added/modified
	- [x] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,697 @@
+# 1.4.0 (2022-08-26)
+
+## Edge Agent
+* Fix bug where Edge Agent is updated without backing image  ( [72e5d648c](https://github.com/Azure/iotedge/commit/72e5d648cc25c05d64dfd52010b7178af772445f) )
+* Fix user creation for edgeAgent and edgeHub  ( [388ec1a34](https://github.com/Azure/iotedge/commit/388ec1a341cf18fce3379c6750e62591963c8624) )
+* Add total memory to device metadata  ( [683a2dde6](https://github.com/Azure/iotedge/commit/683a2dde66fe970e98bc2b187e3419a7e71f1302) )
+* Support feature flag `ModuleUpdateMode`  ( [303b3fdcc](https://github.com/Azure/iotedge/commit/303b3fdcc065c5f6e2e8087e4f92baaafb03b85a) )
+* Update NewtonSoft to 13.0.1  ( [84e883779](https://github.com/Azure/iotedge/commit/84e8837797e84a8e2ae198e42ecffb1c7b7ae546) )
+* Remove docker mode  ( [40824ed28](https://github.com/Azure/iotedge/commit/40824ed28ee6c51eb96c9459513694ae7fc1c018) )
+
```
Comment (damonbarry) [inline CHANGELOG.md:10]:
```suggestion

## Edge Hub
* Fix user creation for edgeAgent and edgeHub  ( [388ec1a34](https://github.com/Azure/iotedge/commit/388ec1a341cf18fce3379c6750e62591963c8624) )
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,697 @@
+# 1.4.0 (2022-08-26)
```
Comment (and-rewsmith) [inline CHANGELOG.md:1]: Did Micah want to add something here?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,697 @@
...
+
+## Edge Agent
```
Comment (and-rewsmith) [inline CHANGELOG.md:3]: Only the one hub change that Damon mentioned?

**Review by and-rewsmith — COMMENTED** [review]
Looks good to me minus the missing hub section.

**Review by and-rewsmith — DISMISSED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (damonbarry) [reply CHANGELOG.md:3]: Yes.

File: CHANGELOG.md (hunk 2 above)
Comment (damonbarry) [reply CHANGELOG.md:1]: We'll do an update later with the additions Micah wants to make. The commit list is correct as-is, so we'll get this merged now.

**Review by damonbarry — APPROVED** [review]

**Pull request merged**